frontend/src/metabase/components/FormCategoryInput/CategoryRadioPicker.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { connect } from "react-redux";
+
+import Radio from "metabase/core/components/Radio";
+
+import Fields from "metabase/entities/fields";
+import { useOnMount } from "metabase/hooks/use-on-mount";
+
+import { State } from "metabase-types/store";
+
+import { CategoryWidgetProps as CategoryWidgetOwnProps } from "./types";
+
+interface CategoryWidgetStateProps {
+  fieldValues: unknown[][];
+}
+
+interface CategoryWidgetDispatchProps {
+  fetchFieldValues: (opts: { id: number }) => void;
+}
+
+interface CategoryWidgetProps
+  extends CategoryWidgetOwnProps,
+    CategoryWidgetStateProps,
+    CategoryWidgetDispatchProps {}
+
+function mapStateToProps(
+  state: State,
+  { fieldInstance }: CategoryWidgetOwnProps,
+) {
+  const fieldValues = Fields.selectors.getFieldValues(state, {
+    entityId: fieldInstance.id,
+  });
+  return {
+    fieldValues,
+  };
+}
+
+const mapDispatchToProps = {
+  fetchFieldValues: Fields.actions.fetchFieldValues,
+};
+
+function CategoryRadioPicker({
+  value,
+  onChange,
+  fieldInstance,
+  fieldValues = [],
+  fetchFieldValues,
+}: CategoryWidgetProps) {
+  useOnMount(() => {
+    if (typeof fieldInstance.id === "number") {
```

Comment:
When it can be false?

Reply:
When it's coming from a native query or from e.g. a custom expression (every dimension have a `field` method and unless a custom column isn't just a copy of a real column, there's no ID to use)